chain/events/observer.go

```diff
@@ -25,7 +25,9 @@ type observer struct {
 	lk        sync.Mutex
 	head      *types.TipSet
 	maxHeight abi.ChainEpoch
-	observers []TipSetObserver
+
+	observerID uint64
+	observers  map[uint64]TipSetObserver
```

Comment:
Some things may care about the order this map in iterated in (mostly tests tho)

Reply:
I could make a `TipSetObservers` type and implement an iterator pattern over it for consistent ordering, would that work?